app/prepare_proposal.go

```diff
@@ -14,11 +14,9 @@ import (
 // generates the data root for the proposal block and passes it back to
 // tendermint via the BlockData.
 func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
-	sdkCtx, err := app.NewProcessProposalQueryContext()
-	if err != nil {
-		panic(err)
-	}
-
+	// create a context using a branch of the state and loaded using the
```

Comment:
[To understand more][non-blocking]
What do you mean by `a branch of the state`?

Reply:
just that under the hood we're calling this function https://github.com/celestiaorg/cosmos-sdk/blob/24903f4a983911f708fbf138a895ea4abc5dc752/store/rootmulti/store.go#L462-L476, which gives us a fork of the state that we operate on without affecting any other branches